dist/jszip-utils.js

```diff
@@ -49,7 +49,18 @@ var createXHR = window.ActiveXObject ?
 
 
 
-JSZipUtils.getBinaryContent = function(path, callback) {
+JSZipUtils.getBinaryContent = function(path, cbo) {
+    //old callback style
+    if(typeof cbo==="function"){
+        var callback = cbo;
+        cbo = {
+            done: function(data){ callback(null, data); },
+            fail: function(err){ callback(err, null); }
+        };
+    }
+    
+    if(typeof cbo!=="object" || typeof cbo.done==="function" || typeof cbo.fail==="function") throw new Error("You must specify a callback.done and callback.fail");
```

Comment:
Looking at this, I think I'd like to make this function return a promise. Given that the promise would fufill the role of `done` and `fail`, I think it would be better to have the options object have properties `{callback, progress}`. 

People who use the callback/errback style can continue to use that, others can use the promise, and `progress` is available to everyone.

Reply:
Yeah that's a better approach